sympy/plotting/plot.py

```diff
@@ -1559,21 +1560,23 @@ def plot(*args, **kwargs):
     for arg in plot_expr:
         expr = arg[0]
         limit = arg[1]
-        frac = fraction(expr) #to find the denominator
-        new_deno = frac[1]
-        if (not new_deno.is_Integer) and (not new_deno.is_Float) and (not new_deno.is_irrational):
-            root_deno = real_roots(new_deno)   #to find the points of discontinuity
+        root_deno = []
+        for exp in denoms(expr):
```

Comment:
use `denoms(expr, x)` where x is `expr.free_symbols().pop()` -- assuming a univariate expression -- and then only denominators that contain the symbol of interest will be returned

Reply:
okay I'll change this